### PR TITLE
Fix: return actual exchange_rates in QueryState request

### PR DESCRIPTION
**Pull request by pr0n00gler**

### CODE REVIEW

File: contracts/anchor_basset_hub/src/contract.rs [hunk 1]
```diff
@@ -355,9 +355,7 @@ fn withdraw_all_rewards(deps: &DepsMut, delegator: String) -> StdResult<Vec<Cosm
     Ok(messages)
 }
 
-/// Check whether slashing has happened
-/// This is used for checking slashing while bonding or unbonding
-pub fn slashing(deps: &mut DepsMut, env: Env) -> StdResult<State> {
+fn update_state(deps: Deps, env: Env) -> StdResult<State> {
```
Comment (oopcode) [inline contracts/anchor_basset_hub/src/contract.rs:358]: update_state here is misleading (no updates actually happen). We should rename this

File: contracts/anchor_basset_hub/src/testing/tests.rs [hunk 2]
```diff
@@ -1856,12 +1856,11 @@ pub fn proper_slashing() {
...
     assert_eq!(2, res.messages.len());
 
-    // expected exchange rate must be more than 0.9
-    let expected_er = Decimal::from_ratio(Uint128::from(1900u64), Uint128::from(2111u64));
+    // expected exchange rate must be equal 0.9
     let ex_rate = State {};
     let query_exchange_rate: StateResponse =
         from_binary(&query(deps.as_ref(), mock_env(), ex_rate).unwrap()).unwrap();
-    assert_eq!(query_exchange_rate.bluna_exchange_rate, expected_er);
+    assert_eq!(query_exchange_rate.bluna_exchange_rate.to_string(), "0.9");
```
Comment (oopcode) [inline contracts/anchor_basset_hub/src/testing/tests.rs:1863]: 0.9 is way too magic

**Review by oopcode — CHANGES_REQUESTED** [review]

File: contracts/anchor_basset_hub/src/testing/tests.rs [hunk 3]
```diff
@@ -1856,8 +1858,7 @@ pub fn proper_slashing() {
     let res = execute(deps.as_mut(), mock_env(), info.clone(), second_bond).unwrap();
     assert_eq!(2, res.messages.len());
 
-    // expected exchange rate must be more than 0.9
-    let expected_er = Decimal::from_ratio(Uint128::from(1900u64), Uint128::from(2111u64));
+    // expected exchange rate must be equal 0.9
```
Comment (swelf19) [inline contracts/anchor_basset_hub/src/testing/tests.rs:1861]: "equal", are you sure? er was 0.9 and after a second bond(1858) it should be a little bigger.

Comment (pr0n00gler) [reply contracts/anchor_basset_hub/src/testing/tests.rs]: Fixed

**Review by oopcode — APPROVED** [review]

**Pull request merged**